src/Compilers/CSharp/Portable/Syntax/CSharpLineDirectiveMap.cs

```diff
@@ -21,69 +21,143 @@ public CSharpLineDirectiveMap(SyntaxTree syntaxTree)
         // Add all active #line directives under trivia into the list, in source code order.
         protected override bool ShouldAddDirective(DirectiveTriviaSyntax directive)
         {
-            return directive.IsActive && directive.Kind() == SyntaxKind.LineDirectiveTrivia;
+            return directive.IsActive && (directive.Kind() is SyntaxKind.LineDirectiveTrivia or SyntaxKind.LineSpanDirectiveTrivia);
```

Comment:
There's a couple references to SyntaxKind.LineDirectiveTrivia in the IDE layer (relates to classification, in ClassifyTrivia and ClassifyPreprocessorDirective) and one to LineDirectiveTriviaSyntax in CSharpSyntaxFacts.TryGetExternalSourceInfo which will probably need looking at.

Reply:
Thanks, added to the test plan.